konduit-serving-core/src/main/java/ai/konduit/serving/configprovider/PipelineRouteDefiner.java

```diff
@@ -212,6 +224,74 @@ public Router defineRoutes(Vertx vertx, InferenceConfiguration inferenceConfigur
+         * {@link ServingConfig#isCreateLoggingEndpoints()} is true.
+         * Returns the last 100 lines of the log file.
+         */
+        if(inferenceConfiguration.getServingConfig().isCreateLoggingEndpoints()) {
+            router.get("/logs")
+                    .handler(ctx -> {
+                try {
+                    ctx.response()
+                            .putHeader(HttpHeaders.CONTENT_TYPE, "text/html")
+                            .end(FileUtils.readFileToString(new ClassPathResource("web/logs_page.html").getFile(), Charset.defaultCharset()));
+                } catch (Exception e) {
+                    ctx.fail(500, e);
+                }
+            });
+
+        /**
+         * Sets up and endpoint to see server logs if {@link ServingConfig#isCreateLoggingEndpoints()}
+         * is true. Returns the number of last few lines determines by the path param
+         * {@code numberOfLastLinesToRead}. If the path param is an invalid integer,
+         * then the whole log file data will be read and returned.
+         */
+            router.get("/logs/:numberOfLastLinesToRead")
+                    .handler(ctx -> {
+                String numberOfLinesString = ctx.pathParam("numberOfLastLinesToRead");
+
+                try {
+                    ctx.response().putHeader(HttpHeaders.CONTENT_TYPE, "text/plain").end(LogUtils.getLogs(Integer.parseInt(numberOfLinesString)));
+                } catch (Exception e) {
+                    if(e instanceof NumberFormatException) {
+                        if("download".equalsIgnoreCase(numberOfLinesString)) {
+                            ctx.response().sendFile(LogUtils.getLogsFile().getAbsolutePath()).end();
+                        } else if("downloadAsZip".equalsIgnoreCase(numberOfLinesString)) {
+                            try {
+                                File zipFile = Paths.get(LogUtils.getLogsDir(), "logs.zip").toFile();
+                                zipFile.createNewFile();
+                                FileOutputStream fos = new FileOutputStream(zipFile);
+                                ZipOutputStream zipOut = new ZipOutputStream(fos);
+
+                                FileInputStream fis = new FileInputStream(LogUtils.getLogsFile());
+                                ZipEntry zipEntry = new ZipEntry("main.log");
+                                zipOut.putNextEntry(zipEntry);
+
+                                byte[] bytes = new byte[1024];
+                                int length;
+                                while((length = fis.read(bytes)) >= 0) {
+                                    zipOut.write(bytes, 0, length);
+                                }
+
+                                fis.close();
+                                zipOut.close();
+                                fos.close();
+
+                                ctx.response().sendFile(zipFile.getAbsolutePath()).end();
+                            } catch (Exception exception) {
```

Comment:
Use try-with-resources here. If anything goes wrong before the `X.close()` calls we'll leave a bunch of open files etc.

Also use buffered input and output streams whenever dealing with files.

Also manual buffer handling in a loop should be unnecessary... things like commons io - IOUtils.copy(InputStream, outputStream)